ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OTargetEncoderParams.scala

```diff
@@ -0,0 +1,125 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.spark.ml.h2o.param
+
+import org.apache.spark.h2o.utils.H2OSchemaUtils
+import org.apache.spark.ml.h2o.features._
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.ml.param.{Param, Params, StringArrayParam}
+import org.apache.spark.sql.types.{DoubleType, StructField, StructType}
+import org.apache.spark.h2o.Frame
+
+trait H2OTargetEncoderParams extends PipelineStage with Params {
+
+  //
+  // Override pipeline stage members
+  //
+  override def transformSchema(schema: StructType): StructType = {
+    val flatSchema = H2OSchemaUtils.flattenSchema(schema)
+    require(getLabelCol() != null, "Label column can't be null!")
```

Comment:
@mn-mikke the point is that validation here has very little to do with transformation of the schema. It is just validation of the parameters and ideally should be done during `H2OTargetEncoderParams` initialisation. I understand that flattening of the schema helps with validation but still validation should be done long before transformation. Maybe we can reuse/pass around flattened schema. I would understand if we did validation of the Schema here.... then just separate private method for this would be enough. But in this case it is validation of parameters and it should be done somewhere else ( constructor). Just wanted to bring your attention .... feel free to ignore or consider just private method with name that highlights that it is params validation(and not a validation of newly created schema).

Reply:
@deil87 I got your point but the parameters may change after the instance is created
```
val targetEncoder = new H2OTargetEncoder()
  .setInputCols(Array("RACE", "DPROS", "DCAPS"))
  .setLabelCol("CAPSULE")
```

Reply:
@mn-mikke Got your point too :) And the order of these methods is not predefined... we probably can't add validation into these methods - too many combinations of possible settings(orders).